src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,277 @@
+            {
+                // Assign the configured defaults
+                messageHandlerOptions.AutoComplete = messagePumpSettings.Options.AutoComplete;
+                messageHandlerOptions.MaxConcurrentCalls = messagePumpSettings.Options.MaxConcurrentCalls ?? messageHandlerOptions.MaxConcurrentCalls;
+
+                Logger.LogInformation("Message pump options were configured instead of Azure Service Bus defaults.");
+            }
+            else
+            {
+                Logger.LogWarning("No message pump options were configured, using Azure Service Bus defaults instead.");
+            }
+
+            return messageHandlerOptions;
+        }
+
+        private async Task<MessageReceiver> CreateMessageReceiverAsync(AzureServiceBusMessagePumpSettings messagePumpSettings)
+        {
+            var rawConnectionString = await messagePumpSettings.GetConnectionStringAsync();
+            var serviceBusConnectionStringBuilder = new ServiceBusConnectionStringBuilder(rawConnectionString);
+
+            MessageReceiver messageReceiver;
+            if (string.IsNullOrWhiteSpace(serviceBusConnectionStringBuilder.EntityPath))
+            {
+                // Connection string doesn't include the entity so we're using the message pump settings
+                if (string.IsNullOrWhiteSpace(messagePumpSettings.EntityName))
+                {
+                    throw new ArgumentNullException(nameof(messagePumpSettings.EntityName), "No entity name was specified while the connection string is scoped to the namespace");
+                }
+
+                messageReceiver = CreateReceiver(serviceBusConnectionStringBuilder, messagePumpSettings.EntityName, messagePumpSettings.SubscriptionName);
+            }
+            else
+            {
+                // Connection string includes the entity so we're using that instead of the message pump settings
+                messageReceiver = CreateReceiver(serviceBusConnectionStringBuilder, serviceBusConnectionStringBuilder.EntityPath, messagePumpSettings.SubscriptionName);
+            }
+
+            Namespace = messageReceiver.ServiceBusConnection?.Endpoint?.Host;
+
+            ConfigurePlugins();
+
+            return messageReceiver;
+        }
+
+        private MessageReceiver CreateReceiver(ServiceBusConnectionStringBuilder serviceBusConnectionStringBuilder, string entityName, string subscriptionName)
+        {
+            EntityPath = entityName;
+
+            if (string.IsNullOrWhiteSpace(subscriptionName) == false)
+            {
+                EntityPath = $"{EntityPath}/subscriptions/{subscriptionName}";
+            }
+
+            var connectionString = serviceBusConnectionStringBuilder.GetNamespaceConnectionString();
+            return new MessageReceiver(connectionString, EntityPath);
+        }
+
+        private void ConfigurePlugins()
+        {
+            var registeredPlugins = DefineServiceBusPlugins();
```

Comment:
I think according our Arcus rules, this variable type has to be specified?

Reply:
Correct!